src/cromshell/submit/command.py

```diff
@@ -0,0 +1,195 @@
+)
+@click.option(
+    "-d",
+    "--dependencies-zip",
+    type=click.Path(exists=True),
+    required=False,
+    help="ZIP file containing workflow source files that are "
+    "used to resolve local imports. This zip bundle will be "
+    "unpacked in a sandbox accessible to this workflow.",
+)
+@click.pass_obj
+def main(config, wdl, wdl_json, options_json, dependencies_zip):
+    """Submit a workflow and arguments to the Cromwell Server"""
+
+    LOGGER.info("submit")
+
+    validate_input(wdl, wdl_json, options_json, dependencies_zip)
+
+    # Check if Cromwell Server Backend works
+    http_utils.assert_can_communicate_with_server(config)
+
+    # Submit workflow to Cromwell Server
+    LOGGER.info("Submitting job to server: %s", cromshellconfig.cromwell_server)
+    request_out = submit_workflow_to_server(
+        wdl, wdl_json, options_json, dependencies_zip, config
+    )
+
+    # Check to make sure that we actually submitted the job correctly
+    # 1. Check for any initial failure by server to accept the job.
+    http_utils.check_http_request_status_code(
+        short_error_message="Failed to Submit Workflow", response=request_out
+    )
+
+    # 2. Check messages from server for workflow problems.
+    # Get our workflow status and job ID:
+    workflow_status = json.loads(request_out.content)
+
+    # 2.A If the status is not `Submitted`, something went wrong:
+    if workflow_status["status"] != "Submitted":
+        log.display_logo(dead_turtle)
+        io_utils.log_error_and_raise_exception(
+            error_source="Cromshell Server",
+            error_source_message=request_out.text,
+            short_error_message="Server reports job was not properly submitted.",
+        )
+
+    # 2.B If the ID is not an ID, something went wrong:
+    if not io_utils.is_workflow_id_valid(workflow_status["id"]):
+        log.display_logo(dead_turtle)
+        io_utils.log_error_and_raise_exception(
+            error_source="Cromshell Server",
+            error_source_message=request_out.text,
+            short_error_message="Did not get a valid ID back. Something went wrong.",
+        )
+
+    # Everything checks out, display success to terminal
+    log.display_logo(io_utils.turtle)
+    io_utils.pretty_print_json(request_out.text)
+
+    # If we get here, we successfully submitted the job and should track it locally:
+    # 1. Create a directory to hold function input files, using server name
+    run_directory = Path(config.config_dir).joinpath(
+        config.local_folder_name, workflow_status["id"]
+    )
+    io_utils.create_directory(run_directory)
+
+    # 2. Copy input to run directory
+    io_utils.copy_files_to_directory(
+        run_directory, [wdl, wdl_json, options_json, dependencies_zip]
+    )
+
+    # 3. Update config.submission_file_path:
+    update_submission_file(
+        config.cromwell_server, config.submission_file_path, wdl, workflow_status
+    )
+
+    return 0
```

Comment:
Originally submit also had a "return_val", but it was suggested to just return 0 because nothing within the code was changing the "return_val" variable.